src/main/java/org/apache/sysml/runtime/instructions/spark/utils/RDDConverterUtilsExt.java

```diff
@@ -362,4 +354,68 @@ private void flushBufferToList( ReblockBuffer rbuff,  ArrayList<Tuple2<MatrixInd
 			ret.addAll(SparkUtils.fromIndexedMatrixBlock(rettmp));
 		}
 	}
+
+	public static Dataset<Row> stringDataFrameToVectorDataFrame(SQLContext sqlContext, Dataset<Row> inputDF)
+			throws DMLRuntimeException {
+
+		StructField[] oldSchema = inputDF.schema().fields();
+		StructField[] newSchema = new StructField[oldSchema.length];
+		for (int i = 0; i < oldSchema.length; i++) {
+			String colName = oldSchema[i].name();
+			newSchema[i] = DataTypes.createStructField(colName, new VectorUDT(), true);
+		}
+
+		// converter
+		class StringToVector implements Function<Tuple2<Row, Long>, Row> {
+			private static final long serialVersionUID = -4733816995375745659L;
+
+			@Override
+			public Row call(Tuple2<Row, Long> arg0) throws Exception {
+				Row oldRow = arg0._1;
+				int oldNumCols = oldRow.length();
+				if (oldNumCols > 1) {
```

Comment:
The idea is that since Vectors type of data is not exported in R they appear as the one String. So only one string is supported as the "[1, 3.4, 3, 123]" or varous sep and format (see the regexp)